src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,96 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private int size = 0;
+    private Object[] array = new Object[10];
+
     @Override
     public void add(T value) {
-
+        if (size + 1 == array.length) {
+            Object[] newArray = new Object[array.length + array.length / 2];
+            System.arraycopy(array, 0, newArray, 0, array.length);
+            array = newArray;
+        }
+        array[size] = value;
+        size++;
     }
 
     @Override
-    public void add(T value, int index) {
-
+    public void add(T value, int index) throws ArrayListIndexOutOfBoundsException {
+        if (index > size() || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index out of bounds");
+        }
+        if (size + 1 >= array.length) {
```

Comment:
The condition `size + 1 >= array.length` should be `size >= array.length` to ensure resizing occurs when the array is full.